### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by JakariaJishan**
- Use webpack to bundle JavaScript.
- use proper ES6 syntax.
- Use ES6 modules to write modular JavaScript.

### CODE REVIEW

File: src/interaction.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  localStorage.setItem('todos', JSON.stringify(todos) || []);
+};
+// clear all completed mark
+
+const clearCompleted = () => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  const completedTodos = todos.filter((item) => item.completed !== true);
+  localStorage.setItem('todos', JSON.stringify(completedTodos) || []);
+  window.location.reload();
+};
```
Comment (Nemwel-Boniface) [inline src/interaction.js]: - [x] When a user deletes all the completed tasks, the values that are remaining are supposed to be saved in the local storage **and their indexes should be updated and start from 1**. Currently, when I have three todos and I finish the first one and delete the first one, I should have an id of 1 and 2 and not 2 and 3 in my local storage. Hope this makes sense.

**This is what the project requirements say:**

image

**This is what I mean that your id's are not updating correctly:**

[Screencast from 26-01-2023 09:39:21 ALASIRI.webm](https://user-images.githubusercontent.com/86318284/214921706-626ce20e-a10e-45a8-8da7-1124b68b8e58.webm)

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @JakariaJishan      :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/interaction.js (hunk 1 above)
Comment (JakariaJishan) [reply src/interaction.js]: thank you , i fix it.

**Review by Deepakdanger — APPROVED** [review]
Hi @JakariaJishan , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-  [x] After marking completed any one to-do, when I refresh the page the marking got unselected, Kindly fix this.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**